### ISSUE TITLE
status of concurrent requests issue?

### ISSUE THREAD

**Issue by nttaylor** [mashpie/i18n-node#176]
In January 2013, John Resig (@jeresig) published a fork of this repository to correct a problem that allegedly could arise when the server received near-simultaneous requests for the page to be rendered in different languages (see http://ejohn.org/blog/a-strategy-for-i18n-and-node/).

I am interested in using your well established, maintained, and documented package, but the above issue concerns me. I have searched for a reference to it here but have not seen it addressed.  Do concurrency problems exist such as the one described by @jeresig?  If so have they been fixed?

**Comment by mashpie** [comment]
that issue resulted from bad usage instructions. People sometimes tried to use i18n in global scope while dealing with req/res circles which never could workout properly as a `locale` setting is always and only bound to a single request. see https://github.com/mashpie/i18n-node#example-usage-in-expressjs

The global use-case refers to an cli-only implementation, as described here https://github.com/mashpie/i18n-node#example-usage-in-global-scope

in 0.4.0 some integration tests where added to cover that issue (https://github.com/mashpie/i18n-node/search?utf8=%E2%9C%93&q=concurrency) as initially described in #4 

So to sum up: there is no concurrency issue, unless wrong implemented :) Take a look at the examples. They are all setup with their own tests covering concurrency as well.

**Comment by millette** [comment]
This rebuttal should probably appear in the README.

I found your module and few others, started using yours (thru req.__(...)) but stumbled on i18n-2 this morning, read Resig's post, searched the commits here for a drastic change sometime in 2013 (when Resig's post was published) and finally found this when I searched the issues for "resig".

Oh, and a big thank you :-)

**Comment by mashpie** [comment]
well, in fact I left a quick comment at johns blog - but it disappeared when comments were closed. But ok, why not document a hint like "How to deal with concurrent request..." - I'll start to write some more detailed walkthrough-example-getting-started best-practice-cookbooks in a separated gitbook in next weeks. Stay tuned and thank you too :)

(any hints, examples, thoughts and proofread welcome - once it started)

**Comment by mashpie** [comment]
...just stumbled about https://github.com/jeresig/i18n-node-2/issues/14
